shortest_path.cc

```diff
@@ -7,7 +8,8 @@
 using namespace std;
 
 namespace dijkstra {
-    const int max_dist = 10;
+    const double max = numeric_limits<double>::max();
```

Comment:
you could use these as local in-function constants (or static const) to avoid linkage problems in future, since if you have a constant called max in other modules.